arch/riscv/core/isr.S

```diff
@@ -60,102 +55,6 @@
-	flr temp, __z_arch_esf_t_ft10_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_ft10_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_ft11_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_ft11_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa0_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa0_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa1_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa1_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa2_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa2_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa3_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa3_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa4_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa4_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa5_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa5_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa6_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa6_OFFSET(to_reg)		;\
-	flr temp, __z_arch_esf_t_fa7_OFFSET(from_reg)		;\
-	fsr temp, __z_arch_esf_t_fa7_OFFSET(to_reg)
-
-#define COPY_ESF(to_reg, from_reg, temp)			\
-	lr temp, __z_arch_esf_t_mepc_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_mepc_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_mstatus_OFFSET(from_reg)	;\
-	sr temp, __z_arch_esf_t_mstatus_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_ra_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_ra_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_tp_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_tp_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_t0_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_t0_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_t1_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_t1_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_t2_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_t2_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_t3_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_t3_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_t4_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_t4_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_t5_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_t5_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_t6_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_t6_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a0_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a0_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a1_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a1_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a2_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a2_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a3_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a3_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a4_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a4_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a5_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a5_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a6_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a6_OFFSET(to_reg)		;\
-	lr temp, __z_arch_esf_t_a7_OFFSET(from_reg)		;\
-	sr temp, __z_arch_esf_t_a7_OFFSET(to_reg)
-
 #define DO_CALLEE_SAVED(op, reg) \
```

Comment:
The action is actually the parameter in the form of the instruction to apply. I have no clear idea how this could be improved but if you think of something please tell me.